pintereach/src/App.js

```diff
@@ -1,29 +1,75 @@
-import React from 'react';
+import React, { useReducer, useState } from "react";
 import { BrowserRouter as Router, Route, Link, Switch } from "react-router-dom";
+import PrivateRoute from "./components/PrivateRoute";
+import { Reducer, appState } from "./reducers/Reducer";
+import { ArticlesContext } from "./contexts/ArticlesContext";
 
 import Login from "./components/Login";
 import Signup from "./components/Signup";
 import Articles from "./components/Articles";
-import PrivateRoute  from "./components/PrivateRoute";
-
-import './App.css';
+import BoardForm from "./components/BoardForm";
+import UserHome from './components/UserHome.js'
 
+import "./App.css";
 
 function App() {
-  
-  return (
+	const [state, dispatch] = useReducer(Reducer, appState);
+
+	//Category state
+	const [category, setCategory] = useState([
+		{
+			id: 2,
+			name: "Retail"
+		},
+		{
+			id: 3,
+			name: "Real Estate"
+		}
+	]);
+
+	console.log("all categories", category);
+
+	function addCategory(c) {
+		const newCategory = {
+			id: Date.now(),
+			name: c.name
+		};
+		setCategory([...category, newCategory]);
+	}
+
+	return (
 		<Router>
 			<div className="App">
-				<h1>Pintereach Project</h1>
-        <Switch>
-          {/* change login path to home */}
-          <Route exact path="/" component={Login} />
-          <Route path="/signup" component={Signup} />
-          {/* change articles component to protected */}
-          <PrivateRoute path="/protected" component={Articles} />
-          {/* <Route component={Login} /> */}
-          {/* <Route component={Articles} /> */}
-				</Switch>
+				<nav className="navbar">
+					<Link className="link" to="/articles">
+						Articles
+					</Link>
+					<Link className="link" to="board">
+						My Board
+					</Link>
+					{window.localStorage.getItem("token") ? null : (
+						<>
+							<Link className="link" to="/">
+								Login
+							</Link>
+							<Link className="link" to="signup">
+								Signup
+							</Link>
+						</>
+					)}
+				</nav>
+
+				<ArticlesContext.Provider value={{ state, dispatch }}>
+					<Switch>
+						<Route exact path="/" component={Login} />
+						<Route path="/signup" component={Signup} />
+						<PrivateRoute path="/board" component={BoardForm} />
+						<PrivateRoute exact path='/home'>
+							<UserHome addCategory={addCategory}/>
```

Comment:
using the reducer and context should eliminate the need to prop drill from the app level